src/commands/Minion/birdhouse.ts

```diff
@@ -0,0 +1,335 @@
+
+		const timeBirdHouseRun = birdhouse.runTime;
+
+		const storePreviousBirdhouse = previousBirdhouseTraps.lastPlaced;
+
+		const prevBirdhouse = storePreviousBirdhouse
+			? birdhouses.find(
+					_birdhouse =>
+						stringMatches(_birdhouse.name, storePreviousBirdhouse) ||
+						stringMatches(_birdhouse.name.split(' ')[0], storePreviousBirdhouse)
+			  )
+			: null;
+
+		const lastPlacedTime: number = previousBirdhouseTraps.birdhouseTime;
+		const difference = currentDate - lastPlacedTime;
+		/* Initiate a cooldown feature for the birdhouses.
+			Allows for a run of birdhouses to only be possible after the
+			previous run's birdhouses have been filled.*/
+		if (prevBirdhouse && difference < prevBirdhouse.waitTime) {
+			throw `Please come back when your birdhouses are full in ${formatDuration(
+				lastPlacedTime + prevBirdhouse.waitTime - currentDate
+			)}!`;
+		}
+
+		let duration: number = timeBirdHouseRun;
+
+		// Reduce time if user has graceful equipped
+		if (hasGracefulEquipped(msg.author.settings.get(UserSettings.Gear.Skilling))) {
+			boostStr.push('10% time for Graceful');
+			duration *= 0.9;
+		}
+
+		let newBank = { ...userBank };
+		let gotCraft = false;
+		if (!prevBirdhouse || msg.flagArgs.nocraft) {
+			const requiredHouse: [string, number][] = Object.entries(birdhouse.houseItemReq);
+			for (const [houseID, qty] of requiredHouse) {
+				if (!bankHasItem(userBank, parseInt(houseID), qty * 4)) {
+					return msg.send(`You don't have enough ${itemNameFromID(parseInt(houseID))}s.`);
+				}
+				newBank = removeItemFromBank(newBank, parseInt(houseID), qty * 4);
+			}
+		} else {
+			if (msg.author.skillLevel(SkillsEnum.Crafting) < birdhouse.craftLvl) {
+				return msg.send(
+					`${msg.author.minionName} needs ${birdhouse.craftLvl} Crafting to make ${birdhouse.name} during the run or write \`${msg.cmdPrefix}birdhouse run ${type} --nocraft\`.`
+				);
+			}
+			gotCraft = true;
+			const requiredLogs: [string, number][] = Object.entries(birdhouse.craftItemReq);
+			for (const [craftID, qty] of requiredLogs) {
+				if (!bankHasItem(userBank, parseInt(craftID), qty * 4)) {
+					return msg.send(`You don't have enough ${itemNameFromID(parseInt(craftID))}.`);
+				}
+				newBank = removeItemFromBank(newBank, parseInt(craftID), qty * 4);
+			}
+		}
+
+		let canPay = false;
+		const requiredSeeds: [string, number][] = Object.entries(birdhouseSeedReq);
+		for (const [paymentID, qty] of requiredSeeds) {
+			if (bankHasItem(userBank, parseInt(paymentID), qty * 4)) {
+				infoStr.push(
+					`You baited the birdhouses with ${qty * 4}x ${itemNameFromID(
+						parseInt(paymentID)
+					)} .`
+				);
+				newBank = removeItemFromBank(newBank, parseInt(paymentID), qty * 4);
+				canPay = true;
+				break;
+			}
+		}
+
+		if (!canPay) {
+			return msg.send(`You don't have enough seeds to bait the birdhouses.`);
+		}
+
+		await msg.author.settings.update(UserSettings.Bank, newBank);
```

Comment:
instead of doing this, you should try to instead construct a bank of items to remove, then just do `msg.author.removeItemsFromBank(bankToRemove.bank)`